src/main/java/teammates/ui/template/AdminActivityLogTableRow.java

```diff
@@ -0,0 +1,160 @@
+package teammates.ui.template;
+
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import teammates.common.util.ActivityLogEntry;
+import teammates.common.util.Const;
+import teammates.common.util.TimeHelper;
+import teammates.common.util.Url;
+
+public class AdminActivityLogTableRow {
+
+    private static final int TIME_TAKEN_EXPECTED = 10000;
+    private static final int TIME_TAKEN_MODERATE = 20000;
+
+    private ActivityLogEntry activityLog;
+
+    public AdminActivityLogTableRow(ActivityLogEntry entry) {
+        activityLog = entry;
+    }
+
+    // --------------- Additional generated fields ---------------
```

Comment:
Same reason as below. In JSP, these are actually fields. Like `data.userHomeLink`.